### ISSUE TITLE
error when `mergeWith` argument is an array

### ISSUE THREAD

**Issue by caikan** [ReactiveX/rxjs#7278]
### Describe the bug

`mergeWith` can accept an `ObservableInput` as an argument, but when the argument is an array (it is a legal ObservableInput), an error will be reported:
`You provided 'xxx' where a stream was expected. You can provide an Observable, Promise, ReadableStream, Array, AsyncIterable, or Iterable.`

### Expected behavior

Treat the array as a normal `ObservableInput`

### Reproduction code

```typescript
import { EMPTY, mergeWith } from 'rxjs';

EMPTY.pipe(mergeWith([1])).subscribe(console.log);
```

### Reproduction URL

https://stackblitz.com/edit/rxjs-aixohl?file=index.ts

### Version

7.8

**Comment by mrgreentm** [comment]
this is an probleam

**Comment by jakovljevic-mladen** [comment]
Fixed in #7281.
